lib/builder/build_plan.go

```diff
@@ -59,9 +60,14 @@ func NewBuildPlan(
 		if parsedStage.From.Alias != "" {
 			if _, ok := aliases[parsedStage.From.Alias]; ok {
 				return nil, fmt.Errorf("duplicate stage alias: %s", parsedStage.From.Alias)
+			} else if _, err := strconv.Atoi(parsedStage.From.Alias); err == nil {
+				// Docker would return `name can't start with a number or contain symbols`
+				return nil, fmt.Errorf("stage alias cannot be a number: %s", parsedStage.From.Alias)
 			}
-			aliases[parsedStage.From.Alias] = true
+		} else {
+			parsedStage.From.Alias = strconv.Itoa(i)
 		}
+		aliases[parsedStage.From.Alias] = true
```

Comment:
Is this still needed?

Reply:
Still need that for the loop starting on line 81.